### PR TITLE
Fix issue #290

### PR DESCRIPTION
**Pull request by marcuslinke**

### CODE REVIEW

File: src/main/java/com/github/dockerjava/core/NameParser.java [hunk 1]
```diff
@@ -5,15 +5,24 @@
...
+import org.apache.commons.lang.builder.EqualsBuilder;
+import org.apache.commons.lang.builder.ToStringBuilder;
+import org.apache.commons.lang.builder.ToStringStyle;
+
 import com.github.dockerjava.api.model.AuthConfig;
 
 public class NameParser {
 
-    private static final Pattern VALID_HEX_PATTERN = Pattern.compile("^([a-f0-9]{64})$");
+    private static final int RepositoryNameTotalLengthMax = 255;
```
Comment (KostyaSha) [inline src/main/java/com/github/dockerjava/core/NameParser.java]: maybe short `RepoNameMaxLength`?

File: src/main/java/com/github/dockerjava/core/NameParser.java [hunk 2]
```diff
@@ -5,15 +5,24 @@
...
+
+    private static final Pattern RepositoryNameComponentRegexp = Pattern.compile("[a-z0-9]+(?:[._-][a-z0-9]+)*");
```
Comment (KostyaSha) [inline src/main/java/com/github/dockerjava/core/NameParser.java]: What is component?

File: src/main/java/com/github/dockerjava/core/NameParser.java (hunk 1 above)
Comment (marcuslinke) [inline src/main/java/com/github/dockerjava/core/NameParser.java]: Variable names are taken from docker sources. I would like to keep them.

File: src/main/java/com/github/dockerjava/core/NameParser.java [hunk 3]
```diff
@@ -36,30 +45,44 @@ public ReposTag(String repos, String tag) {
...
-            name = nameParts[0];
-            if (VALID_HEX_PATTERN.matcher(name).matches()) {
-                throw new InvalidRepositoryNameException(String.format(
-                        "Invalid repository name (%s), cannot specify 64-byte hexadecimal strings", name));
+        @Override
+        public boolean equals(Object obj) {
+            if (obj instanceof ReposTag) {
+                ReposTag other = (ReposTag) obj;
+                return new EqualsBuilder().append(repos, other.repos).append(tag, other.tag).isEquals();
+
```
Comment (KostyaSha) [inline src/main/java/com/github/dockerjava/core/NameParser.java]: remove new line

File: src/main/java/com/github/dockerjava/core/NameParser.java (hunk 1 above)
Comment (KostyaSha) [inline src/main/java/com/github/dockerjava/core/NameParser.java]: :worried:

**Comment by KostyaSha** [comment]
Do you have any generic Object that can parse, store and provide parts of this data in getters like getTag, getHost, getDigest?
```
 "proto://url:port/imagename:tag" (or digest) 
```

**Comment by marcuslinke** [comment]
Sadly no. But as already said in another post all this name handling classes needs to be refactored. It's on my todo list, but I'm quite busy to fix bugs atm.

**Comment by KostyaSha** [comment]
@marcuslinke no problems, all this partial classes periodically confuses TODO rework it is good idea

**Comment by KostyaSha** [comment]
@marcuslinke it maybe good idea squashing such minor changes during merge to have better commit/blame history. I usually comparing docker-api docs, docker-java changes and docker client so it difficult sometimes jumping on styling commits

**Comment by KostyaSha** [comment]
:+1: for change

**Pull request merged**